cumulocity-blocks/CreateMultiMeasurement.mon

```diff
@@ -0,0 +1,150 @@
+/*
+ * $Copyright (c) 2018-2024 Software AG, Darmstadt, Germany and/or Software AG USA Inc., Reston, VA, USA, and/or its subsidiaries and/or its affiliates and/or their licensors.$
+ * Use, reproduction, transfer, publication or disclosure is prohibited except as specifically provided for in your License Agreement with Software AG
+ */
+
+package apamax.analyticskit.blocks.cumulocity;
+
+using apama.analyticsbuilder.BlockBase;
+using apama.analyticsbuilder.Activation;
+using apama.analyticsbuilder.Partition_Broadcast;
+
+using com.apama.cumulocity.Measurement;
+using com.apama.cumulocity.MeasurementValue;
+using com.apama.exceptions.Exception;
+using apama.analyticsbuilder.cumulocity.CumulocityOutputParams;
+using apama.analyticsbuilder.cumulocity.CumulocityOutputHandler;
+using apama.analyticsbuilder.Promise;
+using apama.analyticsbuilder.Value;
+
+event CreateMultiMeasurement_$Parameters {
+	/**
+	 * Device or Trigger Device.
+	 *
+	 * The device (or for models handling groups, Trigger Device) to which the measurement is to be sent.
+	 *
+	 * The model editor uses the current device name. This is mapped internally to the device identifier.
+	 * @$semanticType c8y_deviceIdOrCurrentDevice
+	 */
+	any deviceId;
+
+	/**
+	 * Measurement Type.
+	 *
+	 * The type of the measurement.
+	 */
+	string measurementType;
+
+	/** Validate parameters */
+	action $validate() {
+		switch(deviceId) {
+			case string: { BlockBase.throwsOnEmpty(deviceId, "deviceId", self); }
+			default: {
+				if deviceId = any() or deviceId.getEntry("currentDevice") != <any> true {
+					throw Exception("Expected a device Id or a currentDevice object", "IllegalArgumentException"); // NON-L10N-DEV
+				}
+			}
+		}
```

Comment:
You do not need to do this validation. It is automatically done as part of `CumulocityOutputParams`.

Reply:
will remove. Copied it from here:
https://github.com/SoftwareAG/analytics-builder-blocks-contrib/blob/5a1f3a58192655787aa58a687451cbb6f1240f75/cumulocity-blocks/CreateMeasurement.mon#L59